extensions/src/main/java/org/pentaho/platform/plugin/services/pluginmgr/PlatformPlugin.java

```diff
@@ -72,6 +81,31 @@ public class PlatformPlugin implements IPlatformPlugin, IPentahoInitializer {
 
   private Map<String, List<String>> externalResources = new HashMap<String, List<String>>();
 
+  /**
+   * The regular expression pattern used to match resource bundle keys in the format "${key}".
+   */
+  private final Pattern RESOURCE_BUNDLE_KEY_PATTERN = Pattern.compile( "\\$\\{([^}]+)\\}" );
+
+  /**
+   * The default resource bundle class name.
+   */
+  static final String DEFAULT_RESOURCE_BUNDLE_CLASS_NAME = "messages";
+
+  /**
+   * A fully qualified class name used as the resource bundle base name.
+   * The default value is "messages" and the resource bundle is loaded from the plugin root folder.
+   */
+  private String resourceBundleClassName = DEFAULT_RESOURCE_BUNDLE_CLASS_NAME;
```

Comment:
Better not have a default here, imo. If the bundle does not exist, it will be trying to access for "nothing".

I don't think there is / should be a default `resourcebundle` (also, it is not documented in the proposed `IPlatformPlugin` documentation).